de.tudarmstadt.ukp.wikipedia.datamachine/src/main/java/de/tudarmstadt/ukp/wikipedia/datamachine/dump/xml/SimpleBinaryDumpWriter.java

```diff
@@ -41,31 +45,48 @@
 	private Page currentPage;
 	private Revision lastRevision;
 
+	public SimpleBinaryDumpWriter(DataMachineFiles files) throws IOException {
+		this.files = files;
+		if (this.files.isCompressGeneratedFiles()) {
+			createCompressed();
+		} else {
+			createUncompressed();
+		}
+	}
+
 	protected void createUncompressed() throws IOException {
-		pageFile = new UTFDataOutputStream(new BufferedOutputStream(new FileOutputStream(files.getGeneratedPage())));
-		revisionFile = new UTFDataOutputStream(
-				new BufferedOutputStream(new FileOutputStream(files.getGeneratedRevision())));
-		textFile = new UTFDataOutputStream(new BufferedOutputStream(new FileOutputStream(files.getGeneratedText())));
+		pageFile = openUTFDataOutputStream(files.getGeneratedPage(), false);
+		revisionFile = openUTFDataOutputStream(files.getGeneratedRevision(), false);
+		textFile = openUTFDataOutputStream(files.getGeneratedText(), false);
 	}
 
 	protected void createCompressed() throws IOException {
-
-		pageFile = new UTFDataOutputStream(
-				new GZIPOutputStream(new BufferedOutputStream(new FileOutputStream(files.getGeneratedPage()))));
-		revisionFile = new UTFDataOutputStream(
-				new GZIPOutputStream(new BufferedOutputStream(new FileOutputStream(files.getGeneratedRevision()))));
-		textFile = new UTFDataOutputStream(
-				new GZIPOutputStream(new BufferedOutputStream(new FileOutputStream(files.getGeneratedText()))));
-
+		pageFile = openUTFDataOutputStream(files.getGeneratedPage(), true);
+		revisionFile = openUTFDataOutputStream(files.getGeneratedRevision(), true);
+		textFile = openUTFDataOutputStream(files.getGeneratedText(), true);
 	}
 
-	public SimpleBinaryDumpWriter(DataMachineFiles files) throws IOException {
-		this.files = files;
-		if (this.files.isCompressGeneratedFiles()) {
-			createCompressed();
+	private UTFDataOutputStream openUTFDataOutputStream(final String filePath, boolean compressed) throws IOException {
```

Comment:
Why use `final` on one argument but not on the other? Normally, either use it on all arguments (to underline that Java uses call-by-value) or on none of the arguments.